src/lightning/pytorch/utilities/combined_loader.py

```diff
@@ -77,6 +83,11 @@ def __iter__(self) -> Self:
         self._consumed = [False] * len(self.iterables)
         return self
 
+    def __len__(self) -> int:
+        if self.limits is not None:
+            return max(self.limits)  # type: ignore[return-value]
+        return max(_get_iterables_lengths(self.iterables))  # type: ignore[return-value]
```

Comment:
`len` is expected to return an int, but here we may return a `float("inf")` from the max operation. Ignored here because the only way to call `len(combined_loader)` is with `sized_length`.